src/jdk.hotspot.agent/linux/native/libsaproc/ps_core.c

```diff
@@ -424,6 +424,13 @@ static bool read_lib_segments(struct ps_prochandle* ph, int lib_fd, ELF_EHDR* li
             (existing_map->fd != lib_fd) &&
             (ROUNDUP(existing_map->memsz, page_size) != ROUNDUP(lib_php->p_memsz, page_size))) {
 
+          if (ROUNDUP(existing_map->memsz, page_size) ==
+                ROUNDUP(lib_php->p_memsz + (lib_php->p_offset & (page_size - 1)), page_size)) {
+            // Not an error: the offset in the core dump is aligned down to page_size.
+            // We should trust the core dump in this case.
+            continue;
+          }
```

Comment:
If this added check passes, then you will ignore any error that may have been caught by the previous page_size and lib_fd checks. It think you will need to redo those here before doing the continue. The other choice is to add this extra check to the original `if` condition, and require that just one of the two  `ROUNDUP(existing_map->memsz, page_size)` checks passes.

Reply:
Thanks for pointing that out, I fixed the problem in 5b5940803c373fa5ea000c673cf52625abaaefb8

Let me know if you see any other inconsistencies